nsqlookupd/http.go

```diff
@@ -95,9 +95,31 @@ func (s *httpServer) doChannels(w http.ResponseWriter, req *http.Request, ps htt
 		return nil, http_api.Err{400, "MISSING_ARG_TOPIC"}
 	}
 
+	var producers Producers
+	topicPaused := false
+	channelPausedStatsMap := make(map[string]bool)
 	channels := s.ctx.nsqlookupd.DB.FindRegistrations("channel", topicName, "*").SubKeys()
+	for _, channelName := range channels {
+		if _, ok := channelPausedStatsMap[channelName]; !ok {
+			channelPausedStatsMap[channelName] = false
+		}
+		producers = s.ctx.nsqlookupd.DB.FindProducers("channel", topicName, channelName)
+		producers = producers.FilterByActive(s.ctx.nsqlookupd.opts.InactiveProducerTimeout,
+			s.ctx.nsqlookupd.opts.TombstoneLifetime)
+		if producers.IsChannelPaused() {
+			channelPausedStatsMap[channelName] = true
+		}
+	}
+	producers = s.ctx.nsqlookupd.DB.FindProducers("topic", topicName, "*")
+	producers = producers.FilterByActive(s.ctx.nsqlookupd.opts.InactiveProducerTimeout,
+		s.ctx.nsqlookupd.opts.TombstoneLifetime)
+	if producers.IsTopicPaused() {
+		topicPaused = true
+	}
 	return map[string]interface{}{
-		"channels": channels,
+		"channels":             channels,
+		"channel_paused_stats": channelPausedStatsMap,
+		"topic_paused":         topicPaused,
```

Comment:
This API is more challenging because it's public.

First, it doesn't seem appropriate to return `topic_paused` when the endpoint is `/channels` (regretting not making this a _little_ more RESTful).

Second, it's correct that we'll need to add a new response key for channel paused state, but perhaps what we should do is name it `channel_meta` which could be map of:

```go
type channelMeta struct {
    Paused bool `json:"paused"`
}
```

Which will allow us to add additional properties in the future more easily?